aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -94,6 +94,32 @@ SparseTensor& mul_sparse_scalar_(SparseTensor& t, Scalar v) {
   return mul_out_sparse_scalar(t, t, v);
 }
 
+// --------------------------------------------------------------------
+// log1p(SparseTensor)
+// --------------------------------------------------------------------
+
+// TODO: add in-place variant
+
+SparseTensor& log1p_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  AT_ASSERT(r.is_sparse());
+  AT_ASSERT(t.is_sparse());
+
+  if (isSameTensor(r, t)) {
+    // don't have in-place log1p because coalesce() is not in-place
+    AT_CHECK(r.is_coalesced(), "log1p only applies to coalesced sparse tensor!");
```

Comment:
Not that it "only applies"; that we haven't implemented support for in-place log1p uncoalesced tensors. We want people to submit a bug report if they realize this is something they need.